src/jit/lsra.cpp

```diff
@@ -2563,12 +2562,25 @@ LinearScan::getKillSetForNode(GenTree* tree)
             {
                 needFloatTmpForFPCall = true;
             }
-#endif // _TARGET_X86_
-            killMask = RBM_CALLEE_TRASH;
+        }
+        if (tree->IsHelperCall())
+        {
+            GenTreeCall* call = tree->AsCall();
+            CorInfoHelpFunc helpFunc = compiler->eeGetHelperNum(call->gtCallMethHnd);
+            killMask = compiler->compHelperCallKillSet(helpFunc);
```

Comment:
Here we already know that tree is a GenTreeCall node since its oper is GT_CALL.  Here  too we can use the same condition that was used in codegenxarch.cpp/morph.cpp to know whether a initPInvokeFrame helper call

`if ((call->gtCallType == CT_HELPER) && (call->gtCallMethHnd == compiler->eeFindHelper(CORINFO_HELP_INIT_PINVOKE_FRAME)))`

The it would be 4th such place and assuming that we abstract it in a routine IsInitPInvokeFrameHelper() on GenTreeCall, the above check reduces to

```
if (tree->AsCall()->IsInitPInvokeFrameHelper())
{
     killMask = compiler->compHelperCallKillSet(CORINFO_HELP_INIT_PINVOKE_FRAME);
}
```